projects/vdk-plugins/vdk-impala/src/vdk/plugin/impala/lineage_utils.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2021 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+import logging
+import re
+from typing import Optional
+from typing import Tuple
+
+from impala._thrift_gen.RuntimeProfile.ttypes import TRuntimeProfileFormat
+from impala.hiveserver2 import HiveServer2Cursor
+from vdk.api.lineage.model.sql.model import LineageData
+from vdk.api.lineage.model.sql.model import LineageTable
+
+_log = logging.getLogger(__name__)
+
+
+def get_lineage_data(cursor: HiveServer2Cursor) -> Optional[LineageData]:
+    """
+    Calculates lineage information from the query plan for the last executed query from the provided cursor. Lineage
+    is calculated only if the query has finished successfully and has resulted actual data read or write from/to the
+    underlying storage.
+    """
+    query_profile = cursor.get_profile(profile_format=TRuntimeProfileFormat.STRING)
+    if "Query Status: OK" not in query_profile:
+        return None  # do not capture lineage for failed queries
+    inputs, output = _parse_inputs_outputs(query_profile)
+    if not inputs and not output:
+        return None  # no lineage was present in the query plan, e.g. select 1 query
+    return LineageData(
+        query=cursor._cursor.query_string,  # a bit hacky but otherwise it is not trivial to parse the profile string
+        query_status="OK",
+        query_type=None,  # TODO specify query_type once there is clear spec for it
+        input_tables=[
+            _get_lineage_table_from_table_name(table_name) for table_name in inputs
+        ],
+        output_table=_get_lineage_table_from_table_name(output),
+    )
+
+
+def _parse_inputs_outputs(query_profile: str) -> Tuple[list, str]:
```

Comment:
For which versions of impala would that work ? 
You are using the text profile instead of the thrift but how fixed the schema is for the text profile ? Can we rely on it for future versions?

Reply:
Initially I started with using the thrift format, but that does not seem to be available with the version of impala that is used for testing. Furthermore the interesting part for the lineage is the query plan which is a text field in both string and thrift format, so no difference there.

Reply:
Can we rely on it for future versions? Both can be equally unreliable, but this seems the best we could to.

Reply:
What does the Impala documentation say? If it doesn't you can ask dev@impala.apache.org or user@impala.apache.org Asking never hurts. Worse case they won't answer. 

Even though we cannot predict the future, a good indication we can rely on it is if it has not changed (in breaking way) across multiple major versions of Impala. 

Anyhow this is not an issue to block the change, but let's find out so we know more.